### PR TITLE
[ST] PodSecurityProfilesST, addition of check for KafkaConnect, merge of all of the tests together.

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

- addition of Doc for test 
- merge of test into one
- fix of consumer check with mirror maker 2. In old version client was communicating with source Kafka Cluster.
- addition of testing of Security Profiles on Kafka Connect/Connector. 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java [hunk 1]
```diff
@@ -60,169 +68,158 @@ public class PodSecurityProfilesST extends AbstractST {
...
-        addRestrictedPodSecurityProfileToNamespace(testStorage.getNamespaceName());
 
-        // if Kafka and KafkaBridge Pods deploys it means that it works...
-        resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3).build());
-        resourceManager.createResourceWithWait(extensionContext, KafkaBridgeTemplates.kafkaBridge(testStorage.getClusterName(),
-            KafkaResources.plainBootstrapAddress(testStorage.getClusterName()), 1).build());
+        final String mm1TargetClusterName = testStorage.getTargetClusterName() + "-mm1";
+        final String mm2TargetClusterName = testStorage.getTargetClusterName() + "-mm2";
+        final String mm2SourceMirroredTopicName = testStorage.getClusterName() + "." + testStorage.getTopicName();
+        final int messageCount = 100;
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java:105]: Isn't this count in `Constants`? And also in `testStorage`

File: systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java [hunk 2]
```diff
@@ -60,169 +68,158 @@ public class PodSecurityProfilesST extends AbstractST {
...
-        kafkaClusterAndKafkaMirrorMakerPods.addAll(kubeClient().listPods(testStorage.getNamespaceName(), testStorage.getClusterName(), Labels.STRIMZI_KIND_LABEL, KafkaMirrorMaker2.RESOURCE_KIND));
-
-        // 1. check the generated structure of SecurityContext of Kafka and KafkaMirrorMaker2 Pods
-        // verifies that (i.) Pods and (ii.) Containers has proper generated SC
-        verifyPodAndContainerSecurityContext(kafkaClusterAndKafkaMirrorMakerPods);
-
-        // 2. check that KMM2 mirrors messages
-        resourceManager.createResourceWithWait(extensionContext, kafkaClients.consumerStrimzi());
+        // verifies that (i.) Pods and (ii.) Containers have proper generated SC
+        final List<Pod> kafkaClusterAndKafkaBridgePods = PodUtils.getKafkaClusterPods(testStorage);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java:175]: This is not used anywhere right? Shouldn't it be used in the `verifyPodAndContainerSecurityContext();` on line 181?

Also, I think it's possible that we can collect all "Strimzi pods" using one call, I can check it and help you with that.

File: systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java [hunk 3]
```diff
@@ -60,169 +68,158 @@ public class PodSecurityProfilesST extends AbstractST {
...
-    void testKafkaWithIncorrectConfiguredKafkaClient(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext);
-        addRestrictedPodSecurityProfileToNamespace(testStorage.getNamespaceName());
+        LOGGER.info("Verify that Kafka cluster is usable and everything (MM1, MM2, and Connector) is working");
+        verifyStabilityOfKafkaCluster(extensionContext, testStorage);
 
-        // if Kafka Pods deploys it means that it works...
-        resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3).build());
+        // verify KafkaConnect
+        final String kafkaConnectPodName = kubeClient(testStorage.getNamespaceName()).listPodsByPrefixInName(KafkaConnectResources.deploymentName(testStorage.getClusterName())).get(0).getMetadata().getName();
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java:187]: You can get the podName using the label (as you are doing in few lines above)

**Review by im-konge — COMMENTED** [review]
Few comments, thanks for the changes :)

File: systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java (hunk 1 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java:105]: these are the names of actual Kafka Clusters not mirror makers as such, 
regarding message count i wanted it to be constant 100 hundred regardless of further possible changes in testStorage as i also use the same number for later connector verification.

File: systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java (hunk 2 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java]: great catch i actually passed wrong variable.

**Review by jankalinic — APPROVED** [review]

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --profile=all --install-type=bundle --cluster-type=ocp --cluster-version=4.14 --testcase= PodSecurityProfilesST

**Review by see-quick — APPROVED** [review]

**Review by im-konge — APPROVED** [review]

**Comment by henryZrncik** [comment]
@im-konge  @see-quick  Apologise,  given fail has nothing to do with new test; I simply referenced all tests (most of them failing due to timeout issues none of them caused by these changes)

**Comment by im-konge** [comment]
@strimzi-ci run tests --profile=regression --install-type=bundle --cluster-type=ocp --cluster-version=4.14 --testcase=PodSecurityProfilesST

**Pull request merged**